src/main/scala/akka/contrib/datareplication/ORMap.scala

```diff
@@ -68,23 +82,63 @@ case class ORMap(
    * INTERNAL API
    */
   private[akka] def put(node: UniqueAddress, key: String, value: ReplicatedData): ORMap =
-    ORMap(keys.add(node, key), values.updated(key, value))
+    if (value.isInstanceOf[ORSet] && values.contains(key))
```

Comment:
In general I wrote in the doc:
"Note that the new `value` will be merged with existing values on other nodes and the outcome depends on what `ReplicatedData` type that is used."

That outcome is not always as expected for ORSet, and therefore I wanted to have an early hint that to the user that it is better to use `updated` or the MultiMap. I understand your point, but I will wait with generalizing until requested by something more than ORSet.